cmd/snap/cmd_run.go

```diff
@@ -95,6 +97,38 @@ func (x *cmdRun) Execute(args []string) error {
 		x.Command = "shell"
 	}
 
+	// Check if snapd is starting up. If so wait because it means
+	// snapd is still re-generating security profiles. The
+	// snapd.service is configured so that it will enter a
+	// permanent failed state if it cannot be activated and at this
+	// point the code will continue (with potentially stale profiles
+	// but at least services will run).
+	for i := 0; i < 500; i++ {
```

Comment:
Note that 5 seconds is not very long on armhf. Touch profiles could take as long as 1 second to load, so a system with many profiles on armhf could fallback to old profiles pretty regularly.